tests/field_defaults/tests.py

```diff
@@ -14,3 +34,227 @@ def test_field_defaults(self):
+        headlines = DBArticle.objects.values_list("headline", flat=True)
+        self.assertCountEqual(headlines, expected_headlines)
+
+    def test_bulk_create_all_db_defaults_one_field(self):
+        articles = [
+            DBArticle(pub_date=datetime.now()),
+            DBArticle(pub_date=datetime.now()),
+        ]
+        DBArticle.objects.bulk_create(articles)
+
+        expected_headlines = ["Default headline", "Default headline"]
+        headlines = DBArticle.objects.values_list("headline", flat=True)
+        self.assertCountEqual(headlines, expected_headlines)
+
+    def test_bulk_create_mixed_db_defaults(self):
+        articles = [DBArticle(), DBArticle(headline="Something else")]
+        DBArticle.objects.bulk_create(articles)
+
+        expected_headlines = ["Default headline", "Something else"]
+        headlines = DBArticle.objects.values_list("headline", flat=True)
+        self.assertCountEqual(headlines, expected_headlines)
+
+    @skipUnlessDBFeature(
+        "supports_expression_defaults",
+        "MySQL before 8.0.13 does not support function defaults.",
+    )
+    def test_bulk_create_mixed_db_defaults_function(self):
+        instances = [DBDefaultsFunction(), DBDefaultsFunction(year=2000)]
+        DBDefaultsFunction.objects.bulk_create(instances)
+
+        expected_years = [2000, datetime.now().year]
+        years = DBDefaultsFunction.objects.values_list("year", flat=True)
+        self.assertCountEqual(years, expected_years)
+
+    def test_null_db_default(self):
+        m = DBDefaults.objects.create()
+        if not connection.features.can_return_columns_from_insert:
+            m.refresh_from_db()
+        self.assertEqual(m.null, 1.1)
+
+        m2 = DBDefaults.objects.create(null=None)
+        self.assertIsNone(m2.null)
+
+    @skipUnless(connection.vendor == "postgresql", "Postgres test")
+    def test_both_default_postgres(self):
+        with connection.cursor() as cursor:
+            cursor.execute("INSERT INTO field_defaults_dbdefaults DEFAULT VALUES")
+        m = DBDefaults.objects.get()
+        self.assertEqual(m.both, 2)
+
+        m2 = DBDefaults.objects.create()
+        self.assertEqual(m2.both, 1)
+
+    @skipUnless(connection.vendor == "mysql", "MySQL test")
+    def test_both_default_mysql(self):
+        with connection.cursor() as cursor:
+            cursor.execute("INSERT INTO field_defaults_dbdefaults () VALUES ()")
+        m = DBDefaults.objects.get()
+        self.assertEqual(m.both, 2)
+
+        m2 = DBDefaults.objects.create()
+        self.assertEqual(m2.both, 1)
+
+    @skipUnless(connection.vendor == "sqlite", "Sqlite test")
+    def test_both_default_sqlite(self):
+        with connection.cursor() as cursor:
+            cursor.execute('INSERT INTO field_defaults_dbdefaults ("null") VALUES (1)')
+        m = DBDefaults.objects.get()
+        self.assertEqual(m.both, 2)
+
+        m2 = DBDefaults.objects.create()
+        self.assertEqual(m2.both, 1)
```

Comment:
We should try to avoid using `vendor` in tests as it's not really friendly for 3rd-party backends.

Reply:
What would you recommend instead here? I can't replace the `cursor.execute` with a `DBDefaults.objects.create` call because I'm comparing the behaviour of a raw SQL call to an ORM call. But each database supports a different syntax for the same thing here.

Reply:
The main question is, is it worth keeping these test methods? :thinking: As far as I'm aware, they testing database behavior, not Django behavior.

Reply:
I was thinking of them as confirming whether `db_default` or `default` takes precedence in the two different scenarios. I don't feel strongly about keeping them though.

Reply:
> I was thinking of them as confirming whether `db_default` or `default` takes precedence in the two different scenarios. I don't feel strongly about keeping them though.

But we could confirm this by using `QuerySet` methods without performing raw SQL statements.

Reply:
Can you show an example?

Reply:
> Can you show an example?

Ah right, now I understand. You want to check if the defaults work when someone use raw SQL instead of the ORM, in such case we could add a new attribute to the `DatabaseFeatures`, e.g. on PostgreSQL
```python
insert_test_table_with_defaults = "INSERT INTO %(table_name)s DEFAULT VALUES"
```
see similar test attributes:
https://github.com/django/django/blob/6cbc403b8ee7014bd6dae4892d404eedb1d4a50d/django/db/backends/base/features.py#L261-L268